src/AppBundle/Service/StatsService.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace AppBundle\Service;
+
+use AppBundle\Entity\Report\Report;
+use AppBundle\Entity\User;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\NoResultException;
+use Doctrine\ORM\QueryBuilder;
+
+class StatsService
+{
+    /** @var EntityManager */
+    private $em;
+
+    public function __construct($em)
+    {
+        $this->em = $em;
+    }
+
+    public function countNamedDeputies($type, $from, $to)
```

Comment:
we use type quite extensively in the api, perhaps 'role' would make more sense?

Reply:
and maybe force from and to to be \DateTime 's in case the input changes to a date that doesnt resolve

Reply:
Good points, both done 👍